plans/storm/main.go

```diff
@@ -0,0 +1,251 @@
+
+	var wg syncc.WaitGroup
+	wg.Add(outgoing)
+
+	for outgoing > 0 {
+		randomaddrIdx := rand.Intn(len(otherAddrs))
+
+		addr := otherAddrs[randomaddrIdx]
+
+		outgoing--
+
+		sz := size
+
+		go func() {
+			defer wg.Done()
+
+			delay := time.Duration(rand.Intn(connDelayMs)) * time.Millisecond
+			runenv.RecordMessage("sleeping for: %s", delay)
+			<-time.After(delay)
+
+			sem <- struct{}{}
+
+			t := time.Now()
+			conn, err := net.DialTimeout("tcp", addr, 30*time.Second)
+			if err != nil {
+				runenv.RecordFailure(fmt.Errorf("couldnt dial: %s %w", addr, err))
+				metrics.GetOrRegisterCounter("dial.fail.count", nil).Inc(1)
+				metrics.GetOrRegisterResettingTimer("dial.fail", nil).UpdateSince(t)
+
+				<-sem
+				return
+			}
+			<-sem
+
+			metrics.GetOrRegisterResettingTimer("dial.ok", nil).UpdateSince(t)
+
+			mb := 4 * 1000 // 4kb
+			for sz > 0 {
+				var data []byte
+				if sz <= mb {
+					data = make([]byte, sz)
+					sz = 0
+				} else {
+					data = make([]byte, mb)
+					sz -= mb
+				}
+				rand.Read(data)
+
+				n, err := conn.Write(data)
+				if err != nil {
+					metrics.GetOrRegisterCounter("conn.write.err", nil).Inc(1)
+					runenv.RecordFailure(fmt.Errorf("couldnt write to conn: %s %w", addr, err))
+					return
+				}
+				metrics.GetOrRegisterCounter("bytes.sent", nil).Inc(int64(n))
+			}
+		}()
+	}
+
+	wg.Wait()
+
+	runenv.RecordMessage("done writing")
+	_ = client.MustSignalAndWait(ctx, sync.State("done writing"), runenv.TestInstanceCount)
+	runenv.RecordMessage("done writing after barrier")
+
+	time.Sleep(10 * time.Second)
+
+	return nil
+}
+
+func handleRequest(runenv *runtime.RunEnv, conn net.Conn) {
+	n := -1
+	for n != 0 {
+		buf := make([]byte, 1024*4) // 4kb.
+		var err error
+		n, err = conn.Read(buf)
+		if err != nil && err != io.EOF {
+			metrics.GetOrRegisterCounter("conn.read.err", nil).Inc(1)
+			fmt.Println("Error reading:", err.Error())
+		}
+		metrics.GetOrRegisterCounter("bytes.read", nil).Inc(int64(n))
+		runenv.RecordMessage("read bytes: %d", n)
+	}
+
+	conn.Close()
+}
+
+func getSubnetAddr(subnet *runtime.IPNet) (*net.TCPAddr, error) {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return nil, err
+	}
+	for _, addr := range addrs {
+		if ip, ok := addr.(*net.IPNet); ok {
+			if subnet.Contains(ip.IP) {
+				tcpAddr := &net.TCPAddr{IP: ip.IP}
+				return tcpAddr, nil
+			}
+		} else {
+			panic(fmt.Sprintf("%T", addr))
+		}
+	}
+	return nil, fmt.Errorf("no network interface found. Addrs: %v", addrs)
+}
+
+func shareAddresses(ctx context.Context, client *sync.Client, runenv *runtime.RunEnv, mynodeInfo *ListenAddrs) ([]string, error) {
+	subCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ch := make(chan *ListenAddrs)
+	if _, _, err := client.PublishSubscribe(subCtx, PeerTopic, mynodeInfo, ch); err != nil {
+		return nil, errors.Wrap(err, "publish/subscribe failure")
+	}
+
+	res := []string{}
+
+	for i := 0; i < runenv.TestInstanceCount; i++ {
```

Comment:
@raulk we have to think if we can speed this up. When we have 10k instances, each of them has to read 10k records == 10^10 calls.

Ultimately this operation takes ~5-6min. and I wonder if we can make it faster.